src/UI/Components/Diagram/actions.ts

```diff
@@ -132,7 +131,12 @@ export function showLinkTools(
             ) {
               elementCell.set("embeddedTo", undefined);
               toggleLooseElement(paper.findViewByModel(elementCell), "add");
-              updateInstancesToSend(elementCell, ActionEnum.UPDATE);
+
+              document.dispatchEvent(
+                new CustomEvent("updateInstancesToSend", {
+                  detail: { cell: elementCell, actions: ActionEnum.UPDATE },
+                }),
+              );
               return true;
```

Comment:
I'm really not a fan of this kind of practice, I'm still firmly believing an async, thenable method would be much better, and more readable.

Reply:
removed boolean, added comment and called the function for both directions. I don't agree with async and thenable approach here as there is no need to take it out of the synchronous flow.